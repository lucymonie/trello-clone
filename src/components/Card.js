import React from 'react';

const Card = ({ task }) => (
  <div className="white card-style">
    <span>{task}</span>
  </div>
)

export default Card;
