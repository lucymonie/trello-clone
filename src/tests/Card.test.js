import React from 'react';
import { shallow } from 'enzyme';
import Card from '../components/Card';

describe("Card", () => {
  it("renders consistently", () => {
    const card = shallow(<Card />);
    expect(card).toMatchSnapshot();
  });

  it("renders a task if this is passed in through props", () => {
    const task = "Buy coffee beans";
    const card = shallow(<Card task={task} />);
    expect(card.text()).toBe("Buy coffee beans");
  });

});
