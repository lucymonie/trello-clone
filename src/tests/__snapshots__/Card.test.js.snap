// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span />,
      "className": "white card-style",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span />,
        "className": "white card-style",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
